### PR TITLE
16556/enhancements for wf reports

### PR DESCRIPTION
**Pull request by assuntad23**
Some completed enhancements requested in #16556 
- For: Insert current time as text. Time shows as "16 August 2023 at 06:42", it would be good to specify this is UTC
- For: Time workflow was invoked. Shows as "2023-08-16T06:33:14.822141". Would be good to format more clearly, and specify as UTC
- Useful inclusion: workflow version

## How to test the changes?
(Select all options that apply)
- [X] Instructions for manual testing are as follows:
  1. Create a Workflow
  2. In the edit report view, select the option to add Time a Workflow was invoked
  3. View the workflow report
  4. Note that there is a workflow version at the top, below the title
  5. Note that the current time indicates that it is in UTC
  6. Note that the Time a Workflow was invoked is formatted clearly and indicates it is in UTC

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by mvdbeek** [comment]
> * Useful inclusion: workflow version

I would only use the TRS version if it is available. Otherwise this only corresponds to the user's version, which matches nothing. No version here is probably better than the internal version.

**Comment by jmchilton** [comment]
> I would only use the TRS version if it is available. Otherwise this only corresponds to the user's version, which matches nothing. No version here is probably better than the internal version.

We have a drop down to see old versions of the workflow right? So the number has some value to the end user for tracking down what ran, if the workflow has been updated, etc.. Along those lines - we could display the update time or a link maybe? I don't love displaying just the number... but is it not better than nothing?

Ooooo... maybe just an info statement about whether an updated version of the workflow exists there would be better? ("A newer version of this workflow exists, click here to run it!")

But I guess we should get feedback from the user as to whether they are specifically looking for the TRS version?

**Comment by mvdbeek** [comment]
> We have a drop down to see old versions of the workflow right? 

that's what I added the version for ... i should have called it a checkpoint though :(.

> So the number has some value to the end user for tracking down what ran,

I would think the report is mostly for users that did not run the workflow ?

> but is it not better than nothing?

I worry about it saying version 2, but it's not clear at all this is version 2 of the very specific stored workflow, causing great confusion

Can we just include the link to run the particular version ?

**Comment by jmchilton** [comment]
> that's what I added the version for ... i should have called it a checkpoint though :(.

Let's change it then? Would you be okay with exposing the checkpoint if that is what we called it? 

> I would think the report is mostly for users that did not run the workflow ?

If you pull down a workflow and run it on your data and it has all the results summarized with context and presentation and caveats/imitations/assumptions spelled out  - that is pretty useful for the person running the workflow - a lot more useful than just history items IMO. Especially if they are a Galaxy novice and didn't write the workflow. I intended it initially to be 50/50 - but in practice you need to own all the stuff and such - I think Sam's publishing a report to a page usecase is handling the "not for the user that ran the workflow" better and we really should think about these things as useful things for the workflow user (invoker) potentially. 

> Can we just include the link to run the particular version ?

I like exposing the checkpoint number - but the actual link would be a first priority and more immediately useful thing.

**Comment by mvdbeek** [comment]
yes, sounds good.

**Comment by davelopez** [comment]
FWIW, I think we use `revision` in pages as a similar concept, instead of `version` or `checkpoint`. Just adding my 2 cents :laughing:

**Comment by assuntad23** [comment]
Alright -- so, to recap here: I should instead provide a link to rerun the workflow with the particular version that the user used.
And potentially rename the attribute of the workflow to revision or checkpoint and then expose that on the page as well. Does that sound correct?

File: client/src/components/Markdown/Markdown.vue [hunk 1]
```diff
@@ -22,15 +22,28 @@
                     Edit
                     <FontAwesomeIcon icon="edit" />
                 </b-button>
+                <b-button
+                    v-b-tooltip.hover
+                    class="float-right markdown-edit mr-2"
+                    role="button"
+                    size="sm"
+                    title="Run Workflow Again"
```
Comment (mvdbeek) [inline client/src/components/Markdown/Markdown.vue:30]:
```suggestion
                    title="Run Workflow"
```

File: client/src/components/Markdown/Markdown.vue [hunk 2]
```diff
@@ -22,15 +22,28 @@
...
+                    @click="rerunWorkflow()">
+                    Run Again
+                    <FontAwesomeIcon icon="redo" />
```
Comment (mvdbeek) [inline client/src/components/Markdown/Markdown.vue:33]: This isn't a redo as we do for tools, you gotta select all the parameters again.

I'd also move this into its own component and then share it between the different sports where we run workflows.

**Review by mvdbeek — COMMENTED** [review]
Did you check that these appear in the PDF as well ?
I would be in favor of breaking each feature addressed down into a single PR, so we can merge things as they start looking good.

File: client/src/components/Markdown/Markdown.vue (hunk 2 above)
Comment (assuntad23) [reply client/src/components/Markdown/Markdown.vue:33]: So, when the user clicks this button, they should not have to select an input again? They would want it re-run with the input already selected? 

Is that what you mean?

Comment (assuntad23) [reply client/src/components/Markdown/Markdown.vue:33]: > I'd also move this into its own component and then share it between the different sports where we run workflows.

+1 to moving it to it's own component. Are there other places where we already do something like this? Can you give me a pointer? TIA ! :)

Comment (mvdbeek) [reply client/src/components/Markdown/Markdown.vue:33]: We don't do workflow re-runs, no. It's a complicated thing to do that we'll want to do, but re-run implies all the same options are chosen, yes. Keep it simple, just running the workflow is fine.

**Comment by assuntad23** [comment]
I've split them off into three PRs so far, since the WF rerun will take a little more work, and will likely still rely on getting the workflowID. 
Please check them out (#16757, #16758, and #16760)

**Comment by assuntad23** [comment]
closing this in favor of separate PRs #16757 #16758 #16760

**Pull request closed without merging**